### DISCUSSION TITLE
Attitude data from complimentary Filter

### DISCUSSION THREAD

**Discussion by minisol0175** [bitcraze/discussions#760] [Q&A]
Hello,

In Firmware, in the sensfusion6.c code I can get the attitude data.

![image](https://github.com/bitcraze/discussions/assets/42127467/88a7503e-bda1-4a82-bfe4-6ca64d17e040)

When getting the roll and pitch from this data, it seems that only the gravitational acceleration value is used and not the gyro data.

When flying in a Complimentary Filter environment, is it correct to get roll and pitch without using gyro?

**Comment by knmcguire** [comment]
It is possible to calculate pitch and roll from acceleration data. Gravity always points to the ground so you can see that as an anchor to which those values can reference it to. Yaw though, moves within the plan that is usually orthogonal at the gravity factor so that needs more info. 

Bit of googling I also found [this stack overflow post ](https://stackoverflow.com/questions/7362417/ios-can-i-get-the-pitch-yaw-roll-from-accelerometer-data#:~:text=pitch%20%3D%20atan%20%28x%20%2F%20sqrt%20%28y%5E2%20%2B,PI%20roll%20%3D%20%28roll%20%2A%20180%29%20%2F%20PI )that explains it even better.

**Reply by krichardsson** [reply]
This looks like an AI-generated message, please don't!
